packages/data-point-service/lib/revalidation-store.js

```diff
@@ -0,0 +1,85 @@
+const debug = require('debug')('data-point-service:cache')
+const throttle = require('lodash/throttle')
+
+const MAX_STORE_SIZE = 10000
+const THROTTLE_WAIT = 1000 // cleanup in every 1 seconds
+
+/**
+ * It marks and deletes all the keys that have their ttl expired
+ * @param {Map} store Map Object that stores all the cached keys
+ */
+function clear (store) {
+  const forDeletion = []
+  const now = Date.now()
+
+  for (const entry of store) {
+    // mark for deletion entries that have timed out
+    if (now - entry[1][0] > entry[1][1]) {
+      forDeletion.push(entry[0])
+    }
+  }
+
+  if (forDeletion.length > 0) {
+    debug(`local revalidation flags that timed out: ${forDeletion}`)
+  }
+
+  forDeletion.forEach(key => store.delete(key))
+}
+
+/**
+ * @param {Map} store Map Object that stores all the cached keys
+ * @param {Number} maxStoreSize Size to limit the store, if above this number keys will no longer be saved
+ * @param {String} key key value
+ * @param {Number} ttl time the key will be alive, expressed in milliseconds
+ */
+function add (store, maxStoreSize, key, ttl) {
+  // do not add more than 10000 keys
```

Comment:
this comment assumes that maxStoreSize is set to 10000, but it could be set to anything (and why 10000? what is the reason for choosing that number?

Reply:
really no reason what so ever but just to place a cap. Honestly, I would like to do something smarter here like start deleting the oldest ones, but I rather have that as a new task. for now, I will add this comment to the code and open task to follow up, sounds good?

Reply:
Sounds good